server/services/maps/custom.js

```diff
@@ -0,0 +1,102 @@
+const ValidationError = require("../../errors/validation");
+
+module.exports = class CustomMapService {
+    constructor() { }
+
+    generateLocations(game, settings, starCount, playerCount) {
+        let json;
+
+        try {
+          json = JSON.parse(settings.galaxy.customJSON)
+        }
+        catch (e) {
+          throw new ValidationError('The custom map JSON is malformed.')
+        }
+
+        const locations = [];
+        const playerIndexes = [];
+        //const nameList = new Set()
+        const homeStars = []
+
+        for (const star of json.stars) {
+            //if (nameList.has(star.name)) continue
+
+            if (!this._checkStarProperty(star?.location, 'x', 'number')) continue
```

Comment:
Do these need to be if statements? It'll throw an error if the property check fails.

Reply:
removed those if's